src/quo2/components/code/snippet.cljs

```diff
@@ -1,175 +1,110 @@
+    [:<>
+     [rn/view {:style style/gradient-container}
+      [linear-gradient/linear-gradient
+       {:style  style/gradient
+        :colors [:transparent (style/gradient-color)]}
+       [rn/view {:style style/gradient}]]]
+     child]
+    child))
 
-         ;; Copy button
-         [rn/view
-          {:style {:position :absolute
-                   :bottom   8
-                   :right    8}}
-          [button/button
-           {:icon     true
-            :type     :grey
-            :size     24
-            :on-press on-copy-press}
-           :main-icons/copy]]]))))
+(defn- calc-line-numer-width
+  [font-scale rows-to-show]
+  (let [max-line-digits (-> rows-to-show str count)]
+    (if (= 1 max-line-digits)
+      18 ;; ~ 9 is char width, 18 is width used in Figma.
+      (* 9 max-line-digits font-scale))))
 
-(defn- wrap-renderer-fn
-  [f {:keys [max-lines on-copy-press]}]
-  (fn [^js props]
-    (reagent/as-element [:f> f
-                         {:rows          (.-rows props)
-                          :max-lines     max-lines
-                          :on-copy-press on-copy-press}])))
+(defn- native-renderer
+  [{:keys [rows max-lines on-copy-press]
+    :or   {max-lines ##Inf}}]
+  (let [font-scale          (:font-scale (rn/use-window-dimensions))
+        total-rows          (count rows)
+        number-rows-to-show (min (count rows) max-lines)
+        line-number-width   (calc-line-numer-width font-scale number-rows-to-show)
+        truncated?          (< number-rows-to-show total-rows)
+        rows-to-show-coll   (if truncated?
+                              (as-> rows $
+                                (update $ number-rows-to-show assoc :last-line? true)
+                                (take (inc number-rows-to-show) $))
+                              rows)]
+    [rn/view {:style (style/container)}
+     [rn/view {:style (style/line-number-container line-number-width)}]
+     [rn/view {:style (style/divider line-number-width)}]
+     [mask-view {:apply-mask? truncated?}
+      [code-block
+       {:rows              rows-to-show-coll
+        :line-number-width line-number-width}]]
+     [rn/view {:style style/copy-button}
+      [button/button
+       {:icon                      true
+        :type                      :grey
+        :size                      24
+        :on-press                  on-copy-press
+        :override-background-color (style/button-background-color)}
+       :main-icons/copy]]]))
 
 (defn snippet
   [{:keys [language max-lines on-copy-press]} children]
   [highlighter/highlighter
    {:language          language
-    :renderer          (wrap-renderer-fn
-                        native-renderer
-                        {:max-lines     max-lines
-                         :on-copy-press #(when on-copy-press
-                                           (on-copy-press children))})
-    ;; Default props to adapt Highlighter for react-native.
-    ;;:CodeTag           react-native/View
```

Comment:
During a cleanup this code was commented, it was causing the screen to crash. Since these properties are mandatory for react native, I added them directly to its wrapper, so we don't need to specify them on each future use